src/components/window-shade-button.tsx

```diff
@@ -0,0 +1,34 @@
+import * as React from "react";
+
+import * as css from "./window-shade.sass";
+
+import ContentConfigurations from "../config/content-configurations";
+
+interface IProps {
+  type: string;
+  onClick: () => void;
+}
+
+interface IState { }
+
+export default class WindowShade extends React.Component<IProps, IState> {
+
+  public render() {
+    const { onClick, type } = this.props;
+    const { icon, label, styleClassName } = ContentConfigurations[type];
+    const cssClassNames = [ css.windowShadeToggle, css[styleClassName] ];
+    const iconProps = {
+      width: "50px",
+      fill: "red"
+    };
+    return (
+      <div className={cssClassNames.join(" ")} onClick={onClick}>
+        <span>
+          {icon(iconProps)}
```

Comment:
if you did somewhere higher up:
```
const Icon = icon; // capital letter
```
then you could say:
```
<Icon {...iconProps} />
```
Not a big difference, but it's clear that it's a React component. I would even use capital letter in ContentConfigurations (icon => Icon) so it's clear we're assigning React class, and not some string or inline svg.

Reply:
Yes, that's clearer.